extensions/amp-timeago/0.2/use-intersect.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {useEffect, useRef, useState} from '../../../src/preact';
+
+/**
+ * @return {function}
+ */
+export function useIntersect() {
+  const {1: setEntry} = useState({});
+  const {0: node, 1: setNode} = useState(null);
+  const observer = useRef(
+    new window.IntersectionObserver(entries => {
+      setEntry(entries);
```

Comment:
This seems to only be used to force a re-render. However, there's no way to get the current entries in the component (eg, to see if we're currently visible or invisible). I think this `useIntersect` function has to take a ref, and return a `currentEntries` value.